tailcfg/tailcfg.go

```diff
@@ -1797,8 +1798,26 @@ type MapResponse struct {
 	// Note that this package's type, due its use of a slice and omitempty, is
 	// unable to marshal a zero-length non-nil slice. The control server needs
 	// to marshal this type using a separate type. See MapResponse docs.
+	//
+	// See PacketFilters for the newer way to send PacketFilter updates.
 	PacketFilter []FilterRule `json:",omitempty"`
 
+	// PacketFilters is a method to send incremental packet filter updates to
+	// the client without having to send the entire packet filter on any
+	// changes. The map keys are server-assigned arbitrary strings. The map
+	// values are the new rules for that key, or nil to delete it. The client
+	// then concatenates all the rules together to generate the final packet
+	// filter. Because the FilterRules can only match or not match, the ordering
+	// of filter rules doesn't matter.
```

Comment:
i'm not sure that's true, i think ordering matters for peerCaps

Reply:
If that's the case we have bigger problems because I've never seen us enforce that.

I'm talking about at the FilterRule level. This doesn't affect the ordering of a `CapGrant []CapGrant` within a particular FilterRule match.

Reply:
we can say something about it being ordered based on the names? thats what your implementation does

Reply:
```
	// filter. Because the FilterRules can only match or not match, the ordering
	// of filter rules doesn't matter. (That said, the client generates the file
	// merged packet filter rules by concananting all the packet filter rules
	// sorted by the map key name. But it does so for stability and testability,
	// not correctness. If something needs to rely on that property, something
	// has gone wrong.)
```

?